File: app/src/main/java/cn/ifafu/ifafu/mvp/score_filter/ScoreFilterConstant.java
package cn.ifafu.ifafu.mvp.score_filter;

import java.util.List;

import cn.ifafu.ifafu.data.entity.Score;
import cn.ifafu.ifafu.base.i.IModel;
import cn.ifafu.ifafu.base.i.IPresenter;
import cn.ifafu.ifafu.base.i.IView;
import io.reactivex.Observable;

class ScoreFilterConstant {

    interface Presenter extends IPresenter {

        void onCheck(Score score, boolean checked);

        void updateIES();
    }

    interface View extends IView {

        void setAdapterData(List<Score> list);

        void setIES(String ies);

    }

    interface Model extends IModel {

        Observable<List<Score>> getScoresFromDB(String year, String term);

        void save(Score score);

    }
}

File: app/src/main/java/cn/ifafu/ifafu/mvp/score_filter/ScoreFilterConstant.kt
package cn.ifafu.ifafu.mvp.score_filter

import cn.ifafu.ifafu.base.i.IModel
import cn.ifafu.ifafu.base.i.IPresenter
import cn.ifafu.ifafu.base.i.IView
import cn.ifafu.ifafu.data.entity.Score
import io.reactivex.Observable

class ScoreFilterConstant {

    interface Presenter : IPresenter {

        fun onCheck(score: Score, checked: Boolean)

        fun updateIES()
    }

    interface View : IView {

        fun setAdapterData(list: List<Score>)

        fun setIES(ies: String)

    }

    interface Model : IModel {

        fun getScoresFromDB(year: String, term: String): Observable<List<Score>>

        fun save(score: Score)

    }
}
